krm-functions/lib/fn-runtime/upstream_inventory_test.go

```diff
@@ -0,0 +1,539 @@
+	inventory := NewUpstreamInventory()
+	type object struct {
+		apiVersion string
+		kind       string
+		name       string
+		dummy      string
+		spec       string
+	}
+
+	test1 := []struct {
+		input object
+	}{
+		{
+			input: object{
+				apiVersion: "a",
+				kind:       "b",
+				name:       "c",
+				spec:       "test1",
+				dummy:      "sample",
+			},
+		},
+		{
+			input: object{
+				apiVersion: "d",
+				kind:       "e",
+				name:       "f",
+				spec:       "test2",
+				dummy:      "sample",
+			},
+		},
+	}
+	test2 := []struct {
+		input object
+	}{
+		{
+			input: object{
+				apiVersion: "a",
+				kind:       "b",
+				name:       "c",
+				spec:       "test3",
+				dummy:      "sample",
+			},
+		},
+		{
+			input: object{
+				apiVersion: "d",
+				kind:       "e",
+				name:       "f",
+				spec:       "test4",
+				dummy:      "sample",
+			},
+		},
+	}
+	for _, tt := range test1 {
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+
+		inventory.AddNewResource(&corev1.ObjectReference{
+			APIVersion:      tt.input.apiVersion,
+			Kind:            tt.input.kind,
+			Name:            tt.input.name,
+			Namespace:       tt.input.dummy,
+			FieldPath:       tt.input.dummy,
+			ResourceVersion: tt.input.apiVersion,
+		}, kubeObjectMade)
+	}
+	for _, tt := range test2 {
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+
+		inventory.AddExistingResource(&corev1.ObjectReference{
+			APIVersion:      tt.input.apiVersion,
+			Kind:            tt.input.kind,
+			Name:            tt.input.name,
+			Namespace:       tt.input.dummy,
+			FieldPath:       tt.input.dummy,
+			ResourceVersion: tt.input.apiVersion,
+		}, kubeObjectMade)
+	}
+	diffList, _ := inventory.Diff()
+	if len(diffList.CreateObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToUpdate: -want 0, +got:\n%v", len(diffList.CreateObjs))
+	}
+	if len(diffList.DeleteObjs) != 0 {
+		t.Errorf("TestDiffWithSpecToUpdate: -want 0, +got:\n%v", len(diffList.DeleteObjs))
+	}
+	if len(diffList.UpdateObjs) != 2 {
+		t.Errorf("TestDiffWithSpecToUpdate: -want 2, +got:\n%v", len(diffList.UpdateObjs))
+	}
+	if len(diffList.UpdateConditions) != 2 {
+		t.Errorf("TestDiffWithSpecToUpdate: -want 2, +got:\n%v", len(diffList.UpdateConditions))
+	}
+}
+
+func TestDiffWithSpecToAddCondition(t *testing.T) {
```

Comment:
This case may have been already covered as commented above. If we add expected conditions result to that test this may be deleted. Keeping it does not hurt as well

Reply:
@ganchandrasekaran can you look at this?

Reply:
These condition were developed at different time so is the reason for seperation. 
Merged it all into 1. Thanks